### ISSUE TITLE
Training Time for VLN-3DFF

### ISSUE THREAD

**Issue by Feliciaxyao** [MrZihan/Sim2Real-VLN-3DFF#2]
Thanks for your excellent work! 
Could you please provide the training and fine-tuning times for ETP-VLN-3DFF and ETP-without-Depth? I noticed that the memory usage on a single GPU is not high (7kmb) when running in eval mode. Is this due to the utilization of the CPU?

**Comment by MrZihan** [comment]
Hi, the pre-training (model_step_100000.pt) and fine-tuning (ckpt.iter15000.pth) of ETPNav-without-Depth are very fast and can be completed within one day using 2~4 NVIDIA 3090 GPUs. 
However, the training speed of ETPNav-3DFF is relatively slow. It requires an additional 4 days and over 10,000 steps of training (ckpt.iter25000.pth) on 2~4 GPUs based on the fine-tuned ETPNav-without-Depth (ckpt.iter15000.pth). 
The core issue is that the code for the semantic map and the 3D feature fields during training still needs improvement, as the GPU utilization is currently low. We are refactoring and optimizing the related code (including [HNR-VLN](https://github.com/MrZihan/HNR-VLN)) to make it more user-friendly for researchers.

**Comment by Feliciaxyao** [comment]
Thank you so much for your detailed answers. 
I have learned that the training time for the "train" mode in training ETPNav-3DFF is quite long. I am wondering if the time for the "eval" mode (val_unseen with 11 scenes) also takes long due to the 3D Feature Fields for panoramic perception. Is the batch size for eval and test set to 2 for this model?

**Comment by MrZihan** [comment]
The time for the "eval" mode is relatively long compared to the original ETPNav, taking approximately 3 hours for R2R Val Unseen. The `NUM_ENVIRONMENTS` can be set as needed and does not necessarily have to be 2. 

The primary reason for the slowness of the "eval" mode is the semantic mapping code used from [CM2](https://github.com/ggeorgak11/CM2), which is very slow and performs many operations on the CPU only. The 3D Feature Fields are actually fast (about 80fps for each view, running Feature Fields for each navigation step taking less than 0.2 seconds).

The core issue with the current code is the low GPU utilization and insufficient optimization (mainly related to the semantic map). However, the demand for GPU computational power is quite low (perhaps the training speed difference between a 1080 and a 3090 is not significant). A single GTX 1060 GPU can allow the robot to operate smoothly in a real-world environment.

**Comment by MrZihan** [comment]
We measured that the navigation action prediction time for each step of GTX1060 is less than 2 seconds, as seen in our [demo](https://github.com/MrZihan/Sim2Real-VLN-3DFF/tree/main/demo). 
For R2R Val Unseen, RTX 3090 predicts each step in less than 1 second. Each navigation episode involves approximately 15 navigation steps, resulting in evaluation times of less than 15 seconds per example (with batch size set to 1).
